perf/graaflib/bron_kerbosch_benchmark.cpp

```diff
@@ -0,0 +1,245 @@
+#include <benchmark/benchmark.h>
+#include <graaflib/algorithm/clique_detection/bron_kerbosch.h>
+#include <graaflib/graph.h>
+
+#include <random>
+#include <vector>
+
+namespace {
+
+// Generating random number of vertices (1- 30) for a clique
+std::random_device dev;
+std::mt19937 rng(dev());
+std::uniform_int_distribution<std::mt19937::result_type> random_clique_size(1,
+                                                                            30);
```

Comment:
I am always a little hesitant adding such non deterministic benchmarks, as it makes comparing results difficult. It might still be nice to have since we use it to determine the runtime complexity as well, just something to be aware of.